nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -610,6 +643,123 @@ def get_initial_cache_state(self, batch_size=1, dtype=torch.float32, device=None
 
         return cache_last_channel, cache_last_time
 
+    def change_attention_model(
+        self,
+        self_attention_model: str = None,
+        att_context_size: List[int] = None,
+        update_config: bool = True,
+        device: torch.device = None,
+    ):
+
+        """
+        Update the self_attention_model which changes the positional encoding and attention layers.
+    
+        Args:
+            self_attention_model (str): type of the attention layer and positional encoding
+                'rel_pos': relative positional embedding and Transformer-XL
+                'rel_pos_local_attn': relative positional embedding and Transformer-XL with local attention using
+                    overlapping windows. Attention context is determined by att_context_size parameter.
+                'abs_pos': absolute positional embedding and Transformer
+                If None is provided, the self_attention_model isn't changed. Defauts to None.
+            att_context_size (List[int]): List of 2 ints corresponding to left and right attention context sizes,
+                or None to keep as it is. Defauts to None.
+            update_config (bool): Whether to update the config or not with the new attention model.
+                Defaults to True.
+            device (torch.device): If provided, new layers will be moved to the device.
+                Defaults to None.
+        """
+
+        if att_context_size:
+            att_context_size = list(att_context_size)
+        else:
+            att_context_size = self._cfg.att_context_size
```

Comment:
Do NOT access config during forward pass. Cache the config values into self.xyz and only access that.
If this is called outside forward just once, that is fine.

Reply:
Ok thats good then, btw why is a module maintaining a config under _cfg? You should call it something else possibly cause its specific to modification cfg